src/notation/internal/notationparts.cpp

```diff
@@ -706,7 +706,7 @@ void NotationParts::replaceInstrument(const ID& instrumentId, const ID& fromPart
         return;
     }
 
-    part->setInstrument(InstrumentsConverter::convertInstrument(newInstrument), oldInstrumentInfo.fraction);
+    score()->undo(new Ms::ChangePart(part, new Ms::Instrument(InstrumentsConverter::convertInstrument(newInstrument)), part->name()));
```

Comment:
score()->undo(new Ms::ChangePart(...)) will be called again right after this line (see doSetPartName implementation). Better to use undo only once

Reply:
Ah yes, missed that, agree it makes sense to use just put the one ChangePart command on the undo stack, thanks.